tensorzero_internal/src/clickhouse.rs

```diff
@@ -163,7 +163,11 @@ impl ClickHouseConnectionInfo {
         }
     }
 
-    pub async fn run_query(&self, query: String) -> Result<String, Error> {
+    pub async fn run_query(
+        &self,
+        query: String,
+        parameters: Option<&HashMap<String, String>>,
```

Comment:
This can be a `HashMap<&str, &str>`, since we never need owned strings inside this method. We could go even further and make it a `Vec<(&str, &str)>`, but I'm not sure if that's worth the change in how duplicate query keys

Reply:
so the full type would be `Option<&HashMap<&str, &str>>`?

Reply:
That's right